src/Validation/Validator.php

```diff
@@ -220,13 +220,18 @@ public function setStopOnFailure(bool $stopOnFailure = true)
      *
      * @param array $data The data to be checked for errors
      * @param bool $newRecord whether the data to be validated is new or to be updated.
+     * @param array<string> $fields Fields to validate.
      * @return array<array> Array of failed fields
      */
-    public function validate(array $data, bool $newRecord = true): array
+    public function validate(array $data, bool $newRecord = true, array $fields = []): array
```

Comment:
The public API can't be changed in 5.x

Reply:
We can go 5.next

Reply:
For now this is just to get the feel if there is a better way to do this, what strategy to use and naming for the feature flag / config.

Reply:
> We can go 5.next

We can't do breaking changes even in a new minor. Changing the public method would cause an error if it's overridden. We don't give BC guarantee for only protected overridden methods in minors.

Reply:
Any other ideas we can do to mitigate in 5.x? Until a solution can be found for 6.x?